app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java

```diff
@@ -66,23 +76,7 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
                 }
             });
             bottomNavigation.setSelectedItemId(getIntent().getIntExtra(RES_ITEM, R.id.dashboard));
-            if (getPreferenceManager().getAppOpenCount() != 0
-                    && getPreferenceManager().getAppOpenCount() % PreferenceManager.AUTOTEST_DIALOG_COUNT == 0
-                    && !getPreferenceManager().isAutomaticTestEnabled()
-                    && !getPreferenceManager().isAskAutomaticTestDialogDisabled()) {
-                new ConfirmDialogFragment.Builder()
-                        .withTitle(getString(R.string.Modal_Autorun_Modal_Title))
-                        .withMessage(getString(R.string.Modal_Autorun_Modal_Text))
-                        .withPositiveButton(getString(R.string.Modal_SoundsGreat))
-                        .withNegativeButton(getString(R.string.Modal_NoThanks))
-                        .withNeutralButton(getString(R.string.Modal_DontAskAgain))
-                        .withExtra(AUTOTEST_DIALOG)
-                        .build().show(getSupportFragmentManager(), null);
-            }
-            else if (getPreferenceManager().getAppOpenCount() != 0
-                    && getPreferenceManager().getAppOpenCount() % PreferenceManager.NOTIFICATION_DIALOG_COUNT == 0
-                    && !getPreferenceManager().isNotifications()
-                    && !getPreferenceManager().isAskNotificationDialogDisabled()) {
+            if (notificationManager.shouldShow()) {
```

Comment:
This seems to be incorrect, the type of modals are two, one for notifications and one for autotest

Reply:
Sorry about that, I did a fresh commit with the updated version let me know